src/main/java/org/opensearch/ad/InsightsJobProcessor.java

```diff
@@ -0,0 +1,643 @@
+
+        if (jobParameter.getSchedule() instanceof IntervalSchedule) {
+            IntervalSchedule intervalSchedule = (IntervalSchedule) jobParameter.getSchedule();
+            long intervalAmount = intervalSchedule.getInterval();
+            ChronoUnit intervalUnit = intervalSchedule.getUnit();
+            executionStartTime = executionEndTime.minus(intervalAmount, intervalUnit);
+            log
+                .info(
+                    "One-time Insights job analyzing data from {} to {} (interval: {} {})",
+                    executionStartTime,
+                    executionEndTime,
+                    intervalAmount,
+                    intervalUnit
+                );
+        } else {
+            log
+                .warn(
+                    "Unexpected schedule type for Insights job {} in one-time run: {}, defaulting to 24 hours",
+                    jobName,
+                    jobParameter.getSchedule() != null ? jobParameter.getSchedule().getClass() : "null"
+                );
+            executionStartTime = executionEndTime.minus(24, ChronoUnit.HOURS);
+            log.info("One-time Insights job analyzing data from {} to {} (default 24h window)", executionStartTime, executionEndTime);
+        }
+
+        querySystemResultIndex(jobParameter, null, null, executionStartTime, executionEndTime);
+    }
+
+    /**
+     * Release lock for job.
+     */
+    private void releaseLock(Job jobParameter, LockService lockService, LockModel lock) {
+        if (lockService == null || lock == null) {
+            log.debug("No lock to release for Insights job {}", jobParameter.getName());
+            return;
+        }
+        lockService
+            .release(
+                lock,
+                ActionListener.wrap(released -> { log.info("Released lock for Insights job {}", jobParameter.getName()); }, exception -> {
+                    log.error(new ParameterizedMessage("Failed to release lock for Insights job {}", jobParameter.getName()), exception);
+                })
+            );
+    }
+
+    /**
+     * Execute the main Insights job logic.
+     * 
+     * @param jobParameter The insights job
+     * @param lockService Lock service for releasing lock
+     * @param lock The acquired lock
+     * @param executionStartTime Start of analysis window
+     * @param executionEndTime End of analysis window
+     */
+    private void runInsightsJob(
+        Job jobParameter,
+        LockService lockService,
+        LockModel lock,
+        Instant executionStartTime,
+        Instant executionEndTime
+    ) {
+        String jobName = jobParameter.getName();
+        if (lock == null) {
+            log.warn("Can't run Insights job due to null lock for {}", jobName);
+            return;
+        }
+
+        log.info("Running Insights job for time window: {} to {}", executionStartTime, executionEndTime);
+
+        querySystemResultIndex(jobParameter, lockService, lock, executionStartTime, executionEndTime);
```

Comment:
Only querying system result index would hinder your ability to go GA alone. You have to tie insights with Auto AD creation. One route to go to GA is to add a text box in AD overview page. That would add a summary on top of existing detectors' results.